src/js/server/utils.js

```diff
@@ -0,0 +1,188 @@
+    var salt        = gpii.express.user.password.generateSalt(that.options.saltLength);
+    var derived_key = gpii.express.user.password.encode(userData.password, salt);
+    var code        = gpii.express.user.password.generateSalt(that.options.verifyCodeLength);
+
+    // Our rules will set the defaults and pull approved values from the original submission.
+    var combinedRecord                   = fluid.model.transformWithRules(userData, that.options.rules.createUserWrite);
+
+    // Set the "name" to the username for backward compatibility with CouchDB
+    combinedRecord.salt                  = salt;
+    combinedRecord.derived_key           = derived_key;
+    combinedRecord[that.options.codeKey] = code;
+
+    // Set the ID to match the CouchDB conventions, for backward compatibility
+    combinedRecord._id = "org.couch.db.user:" + combinedRecord.username;
+
+    // Write the record to couch.  TODO: Migrate this to a writable dataSource.
+    var writeOptions = {
+        url:    that.options.writeUrl,
+        method: "POST",
+        json:   true,
+        body:   combinedRecord
+    };
+    var promiseTogo = fluid.promise();
+    request(writeOptions, function (error, response, body) {
+        if (error) {
+            promiseTogo.reject({isError: true, message: error});
+        }
+        else if ([200, 201].indexOf(response.statusCode) === -1) {
+            promiseTogo.reject({ isError: true, message: body});
+        }
+        else {
+            promiseTogo.resolve(combinedRecord);
+        }
+    });
+    return promiseTogo;
+};
+
+/**
+ * gpii.express.user.utils.verifyPassword function
+ *
+ * Given an instance of our standard couch `userRecord`, and the clear text
+ * `password`, check to see if the password is valid for for the user.
+ *
+ * @param {Object} userRecord - Our standard internal user object stored in CouchDB.
+ * @param {String} password - Password to use to login/unlock user.
+ * @return {Boolean} - True if this is the correct password, otherwise false.
+ */
+gpii.express.user.utils.verifyPassword = function (userRecord, password) {
+    var encodedPassword = gpii.express.user.password.encode(password,
+        userRecord.salt, userRecord.iterations, userRecord.keyLength, userRecord.digest);
+    return encodedPassword === userRecord.derived_key;
+};
+
+/**
+ * gpii.express.user.utils.unlockUser method
+ *
+ * Attempts to look up username using the current CouchDB view. (At the time
+ * of writing either username or email). And unlock their account using `password`.
+ * If the username and password match, the CouchDB `userData` record will be returned.
+ * Otherwise a standard error Object is returned.
+ *
+ * @param {gpii.express.user.utils} that - Utils component.
+ * @param {String} username - Username to use for record lookup.
+ * @param {String} password - Clear text password to validate record with.
+ * @return {Object} The `userData` record if the password is correct, otherwise
```

Comment:
Good catch.  I made a comment about the docs for this, but was wondering if we should get in the habit of explicitly referring to them as `fluid.promise` instances in the JSDocs as well, so that people have an inkling that they are somehow different than ES6 promises or any other variant.

Reply:
Hmm, good idea.  I do enjoy putting things more specific than `Object` in the curly brackets.  I'll use `fluid.promise`

Reply:
This should be updated now.